File: apps/api/app/schemas/product.py
from pydantic import BaseModel


class ProductCreate(BaseModel):
    product_name: str
    category: str
    price: str | None = None
    target_audience: str | None = None
    key_benefits: str | None = None
    pain_points: str | None = None
    usp: str | None = None
    competitor_or_alternative: str | None = None
    selling_intensity: str = "balanced"
    platform: str = "tiktok"
    duration_seconds: int | None = None
    cta: str | None = None
    compliance_notes: str | None = None
    language: str = "vi"
    tone: str = "friendly"


class ProductResponse(ProductCreate):
    id: str

File: apps/web/types/product.ts
export type ProductInput = {
  product_name: string;
  category: string;
  price?: string;
  target_audience?: string;
  key_benefits?: string;
  pain_points?: string;
  usp?: string;
  competitor_or_alternative?: string;
  selling_intensity?: string;
  platform?: string;
  duration_seconds?: number | null;
  cta?: string;
  compliance_notes?: string;
  language?: string;
  tone: string;
};
